ext/dcos-installer/dcos_installer/cli_dispatcher.py

```diff
@@ -0,0 +1,168 @@
+
+
+def log_warn_only():
+    """Drop to warning level and down to get around gen.generate() log.info
+    output"""
+    coloredlogs.install(
+        level='WARNING',
+        level_styles={
+            'warn': {
+                'color': 'yellow'
+            },
+            'error': {
+                'color': 'red',
+                'bold': True,
+            },
+        },
+        fmt='%(message)s',
+        isatty=True
+    )
+
+
+def print_validation_errors(messages):
+    log.error("Validation of configuration parameters failed: ")
+    for key, message in messages.items():
+        log.error('{}: {}'.format(key, message))
+
+
+def validate_ssh_config_or_exit():
+    validation_errors = backend.do_validate_ssh_config()
+    if validation_errors:
+        print_validation_errors(validation_errors)
+        sys.exit(1)
+
+
+def dispatch_option(args):
+    """Dispatches ClI options which are not installer actions"""
+    def web(args):
+        print_header("Starting DC/OS installer in web mode")
+        async_server.start(args)
+
+    def hash_password(args):
+        if len(args.hash_password) > 0:
+            print_header("HASHING PASSWORD TO SHA512")
+            backend.hash_password(args.hash_password)
+            return 0
+
+    def validate_config(args):
+        print_header('VALIDATING CONFIGURATION')
+        log_warn_only()
+        validation_errors = backend.do_validate_gen_config()
+        if validation_errors:
+            print_validation_errors(validation_errors)
+            return 1
+        return 0
+
+    def genconf(args):
+        print_header("EXECUTING CONFIGURATION GENERATION")
+        code = backend.do_configure()
+        if code != 0:
+            return 1
+        return 0
+
+    for action in ['web', 'hash_password', 'validate_config', 'genconf']:
+        if getattr(args, action):
+            sys.exit(locals()[action](args))
+
+
+def dispatch_action(args):
+    """Dispatches CLI options which are installer actions ran through
+    AIO loop construct"""
+    validate_ssh_config_or_exit()
+
+    def preflight(args):
+        print_header("EXECUTING PREFLIGHT")
+        return action_lib.run_preflight
+
+    def postflight(args):
+        print_header("EXECUTING POSTFLIGHT")
+        return action_lib.run_postflight
+
+    def deploy(args):
+        print_header("EXECUTING DC/OS INSTALLATION")
+        return action_lib.install_dcos
+
+    def uninstall(args):
+        print_header("EXECUTING UNINSTALL")
+        tall_enough_to_ride()
+        return action_lib.uninstall_dcos
+
+    def install_prereqs(args):
+        print_header("EXECUTING INSTALL PREREQUISITES")
+        return action_lib.install_prereqs
+
+    for action in ['deploy', 'preflight', 'postflight', 'uninstall', 'install_prereqs']:
+        if getattr(args, action):
+            errors = run_loop(locals()[action](args), args)
+            installer_analytics.send(
+                action=action,
```

Comment:
How about `"installer_{}".format(action)`? I'd like to make sure we're not gonna clash on things like "deploy" in the future.